server/neuserver.cpp

```diff
@@ -208,16 +208,30 @@ void handleHTTP(websocketpp::connection_hdl handler) {
             resource = documentRoot + resource;
         }
     }
-    router::Response routerResponse = router::serve(resource);
-    con->set_status(routerResponse.status);
-    con->set_body(routerResponse.data);
-    con->replace_header("Content-Type", routerResponse.contentType);
+    
+    // Check if the requested resource is not found and if it's not an API request
+    if (!filesystem::exists(resource) && !regex_match(resource, regex("^/api/.*"))) {
+        // Serve the index.html for SPA routing
+        string indexPath = documentRoot + "/index.html"; // Adjust the path as needed
+        ifstream file(indexPath);
+        stringstream buffer;
+        buffer << file.rdbuf();
+        con->set_body(buffer.str());
```

Comment:
Could you please use `fs::readFile()` instead of using C++ standard library directly? :tada: